src/controllers/productController.js

```diff
@@ -50,4 +53,114 @@ const CreateProduct = async (req, res) => {
   }
 };
 
+export const updateProductAvailability = async (req, res) => {
+  try {
+    const { id } = req.params;
+    const product = await Product.findOne({ where: { id } });
+    if (!product) {
+      return res.status(400).json({ status: 400, success: false, message: 'Product not found' });
+    }
+
+    if (product.seller_id !== req.user.id) {
+      return res.status(401).json({ status: 401, success: false, message: 'Unauthorized access!' });
+    }
+
+    // toggle availability
+    const newAvailability = !product.availability;
+
+    await Product.update({ availability: newAvailability }, { where: { id } });
+
+    res.status(200).json({
+      status: 200,
+      success: true,
+      message: 'Product availability updated',
+      data: { id, availability: newAvailability }
+    });
+  } catch (error) {
+    res.status(500).json({ status: 500, success: false, message: error.message });
+  }
+};
+
+export const updateProduct = async (req, res) => {
+  try {
+    const updateData = req.body;
+    if (!Object.keys(updateData).length) {
+      return res.status(400).json({
+        status: 400,
+        success: false,
+        message: 'No data provided',
+      });
+    }
+    const { id } = req.params;
+    const product = await Product.findOne({ where: { id } });
+    if (!product) {
+      return res.status(400).json({ status: 400, success: false, message: 'Product not found in your collection!' });
+    }
+    if (product.seller_id !== req.user.id) {
+      return res.status(401).json({ status: 401, success: false, message: 'Unauthorized access!' });
+    }
+
+    const updatedProduct = await product.update(updateData, { returning: true });
+    if (req.files) {
+      const promises = req.files.map((file) => cloudinary.uploader.upload(file.path, {
+        folder: 'Falcons_E-comm_App/ProductImages',
+        public_id: `${product.id}_image`,
+      }),);
+
+      const results = await Promise.all(promises);
+      updateData.images = results.map((result) => result.url).filter((url) => url);
+    }
```

Comment:
I'm afraid that's not right you updating the product before saving the image first save the image then use returned  imageLink and update it in db if it's there